clients/js/src/helpers/fetch.ts

```diff
@@ -147,6 +147,46 @@ export const fetchAsset = async (
   return deriveAssetPlugins(assetV1, await fetchCollectionV1(umi, collection));
 };
 
+/**
+ * Helper function to fetch multiple assets and derive plugins from their collections if applicable.
+ *
+ * @param umi Context
+ * @param assets Array of asset addresses to fetch
+ * @param options Options, `skipDerivePlugins` plugins from collection is false by default
+ * @returns Promise of a list of `AssetV1`
+ */
+export const fetchAllAssets = async (
+  umi: Context,
+  assets: Array<PublicKey | string>,
+  options: { skipDerivePlugins?: boolean } & RpcGetAccountOptions = {}
+): Promise<AssetV1[]> => {
+  const assetV1s = await Promise.all(
+    assets.map((asset) => fetchAssetV1(umi, publicKey(asset)))
+  );
+
+  if (options.skipDerivePlugins) {
+    return assetV1s;
+  }
+
+  const collectionKeys = Array.from(
+    new Set(assetV1s.map((asset) => collectionAddress(asset)))
+  ).filter((collection): collection is PublicKey => !!collection);
+
+  const collections = await fetchAllCollectionV1(umi, collectionKeys);
+
+  return assetV1s.map((assetV1) => {
+    const collection = collectionAddress(assetV1);
+    if (!collection) {
+      return assetV1;
+    }
+
+    const matchingCollection = collections.find(
```

Comment:
minor performance nit: can we convert the collection list into a map keyed off pubkey to avoid n^2

Reply:
fixed